### PR TITLE
Skip tests for nobody if necessary

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -715,8 +718,11 @@ if id_result.returncode() == 0
                 message('WARNING:\n' +
                         '        The local user with the configured user name "@0@" of the nobody user does not have UID 65534 (it has @1@).\n'.format(nobody_user, id) +
                         '        Your build will result in an user table setup that is incompatible with the local system.')
+                nobody_group_mismatch = true
```
Comment (evverx) [inline meson.build]: This doesn't seem to be correct. `id -u` above checks whether `nobody_user` has a certain id, so it has nothing to do with groups. The tests are likely to still be failing.

Comment (evverx) [reply meson.build]:
```sh
$ meson -Dnobody-user=nobody -Dnobody-group=nobody build && ninja -C ./build
...
Message: WARNING:
        The local user with the configured user name "nobody" of the nobody user does not have UID 65534 (it has 99).
        Your build will result in an user table setup that is incompatible with the local system.
Message: WARNING:
        The local group with the configured group name "nobody" of the nobody group does not have UID 65534 (it has 99).
        Your build will result in an group table setup that is incompatible with the local system.
...

$ grep -r MISMATCH build/config.h
#define NOBODY_GROUP_MISMATCH 1
#define NOBODY_USER_MISMATCH 0

$ sudo touch /etc/systemd/dont-synthesize-nobody

$ sudo ./build/test-user-util
/* test_uid_to_name_one(0, "root") */
/* test_uid_to_name_one(65534, "nobody") */
/* test_uid_to_name_one(65535, "65535") */
/* test_uid_to_name_one(4294967295, "4294967295") */
/* test_gid_to_name_one(0, "root") */
/* test_gid_to_name_one(65534, "nobody") */
/* test_gid_to_name_one(5, "tty") */
/* test_gid_to_name_one(65535, "65535") */
/* test_gid_to_name_one(4294967295, "4294967295") */
/* test_get_user_creds_one("root", "root", 0, 0, "/root", "/bin/sh") */
got "root", 0, 0, "/root", "/bin/sh"
/* test_get_user_creds_one("0", "root", 0, 0, "/root", "/bin/sh") */
got "root", 0, 0, "/root", "/bin/sh"
/* test_get_user_creds_one("nobody", "nobody", 65534, 65534, "/", "/sbin/nologin") */
got "nobody", 99, 99, "/", "/sbin/nologin"
Assertion 'ruid == uid' failed at ../src/test/test-user-util.c:181, function test_get_user_creds_one(). Aborting.
Aborted
```

Comment (keszybz) [reply meson.build]: <strike>I don't see how that can be. You get two warnings, but only NOBODY_GROUP_MISMATCH is set. Did you paste from the wrong build?</strike>

**Comment by keszybz** [comment]
@evverx sorry, I forgot to push one commit yesterday that is necessary if a nobody user is already defined with a matching uid but a different home or shell. So the PR you were testing was different than what I had tested locally.

I now updated the setting of  the two variables to set `nobody_user_mismatch` in the first two checks and `nobody_group_mismatch` in the second two checks. Can you test again?

**Comment by keszybz** [comment]
Meh, I now pushed a simpler version that skips all checks for nobody if synthentization is off. This should be much more robust. Please test.

**Comment by evverx** [comment]
Thank you. It is not only more robust, but also easier to understand. I think it would also be great to print something when the tests are skipped so that the output won't be confusing as it seems to be now:
```
/* test_get_group_creds_one("nobody", "nobody", 65534) */
got "nobody", 99: Success
/* test_get_group_creds_one("65534", "nobody", 65534) */
got "nobody", 65534: Success
```

**Comment by keszybz** [comment]
Updated:
```console
$build/test-user-util
/* test_uid_to_name_one(0, "root") */
/* test_uid_to_name_one(65534, "nobody") */
(skipping detailed tests because nobody is not synthesized)
/* test_uid_to_name_one(65535, "65535") */
/* test_uid_to_name_one(4294967295, "4294967295") */
/* test_gid_to_name_one(0, "root") */
/* test_gid_to_name_one(65534, "nobody") */
(skipping detailed tests because nobody is not synthesized)
/* test_gid_to_name_one(5, "tty") */
/* test_gid_to_name_one(65535, "65535") */
/* test_gid_to_name_one(4294967295, "4294967295") */
/* test_get_user_creds_one("root", "root", 0, 0, "/root", "/bin/sh") */
got "root", 0, 0, "/root", "/bin/sh": Success
/* test_get_user_creds_one("0", "root", 0, 0, "/root", "/bin/sh") */
got "root", 0, 0, "/root", "/bin/sh": Success
/* test_get_user_creds_one("nobody", "nobody", 65534, 65534, "/", "/sbin/nologin") */
got "nobody", 4294967295, 4294967295, "(null)", "(null)": No such process
(skipping detailed tests because nobody is not synthesized)
/* test_get_user_creds_one("65534", "nobody", 65534, 65534, "/", "/sbin/nologin") */
got "65534", 4294967295, 4294967295, "(null)", "(null)": No such process
(skipping detailed tests because nobody is not synthesized)
/* test_get_group_creds_one("root", "root", 0) */
got "root", 0: Success
/* test_get_group_creds_one("0", "root", 0) */
got "root", 0: Success
/* test_get_group_creds_one("nobody", "nobody", 65534) */
got "nobody", 4294967295: No such process
(skipping detailed tests because nobody is not synthesized)
/* test_get_group_creds_one("65534", "nobody", 65534) */
got "65534", 4294967295: No such process
(skipping detailed tests because nobody is not synthesized)
/* test_parse_uid */
/* test_uid_ptr */
/* test_valid_user_group_name */
/* test_valid_user_group_name_or_id */
/* test_valid_gecos */
/* test_valid_home */
```

**Pull request merged**